lib/rdoc/markup/heading.rb

```diff
@@ -66,7 +66,13 @@ def label context = nil
   # element.
 
   def plain_html
-    self.class.to_html.to_html(text.dup)
+    text = self.text.dup
+
+    if text.match?(/rdoc-image:[^:]+:(.*)/)
+      text = text.match(/rdoc-image:[^:]+:(.*)/)[1]
+    end
```

Comment:
```suggestion
    if matched = text.match(/rdoc-image:[^:]+:(.*)/)
      text = matched[1]
    end
```

Reply:
Can we also add a test case in `test/rdoc/test_rdoc_markup_heading.rb`?

Something like:

```rb
  def test_plain_html_using_image_alt_as_text
    h = RDoc::Markup::Heading.new 1, 'rdoc-image:foo.png:Hello World'

    assert_equal 'Hello World', h.plain_html
  end
```

Reply:
If the intention is to remove image and replace it with alt text, `# foo ![alt](a.jpg) bar` → `foo alt bar`
I think gsub is better, and need to consider rdoc-image without alt text.

```
Text1 rdoc-image:img.jpg:alt_text text2 rdoc-image:img_without_alt.jpg text3
↓
Text1 alt_text text2 text3
```

Reply:
@tompng I don't understand how to action your suggestion.